### PR TITLE
Additional support for cypress 10

### PR DESCRIPTION
**Pull request by SophieDePaula**
- [x] add support for `cypress.config.ts` (`pluginExports`, `eyes-setup`)
- [x]  add support for component testing (`pluginExports`, tests, `eyes-setup`)
- [x] update README to point to `index.d.ts` for `ts` rather than `eyes-index.d.ts`
- [x] Test Components
- [x] Test using `cypress.config.ts` in setup
- [x] Test for `cypress.config.ts` file in `pluginExports`

### CODE REVIEW

**Comment by AndriiTsok** [comment]
Hello @SophieDePaula, 
I am wondering if there are any ETAs on this PR? We are facing some issues with cypress.config.ts and Applitools server set up in typescript and component testing variant. Thank you.

**Comment by SophieDePaula** [comment]
> Hello @SophieDePaula, I am wondering if there are any ETAs on this PR? We are facing some issues with cypress.config.ts and Applitools server set up in typescript and component testing variant. Thank you.

@AndriiTsok I hope to get this done in the next few weeks. To configure the setup file manually you can follow our instructions [here](https://www.npmjs.com/package/@applitools/eyes-cypress).
Component support will be added after the work on this PR is done.
If you have any additional questions you can reach out to [our support team](https://help.applitools.com/hc/en-us/requests/new)

File: pending-changes.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ Apollo SDK:
     - Fixed issue with logs not being saved/written
 "@applitools/eyes-cypress":
   feature:
-    - 
+    - Add support for componenet testing and cypress config ts file
```
Comment (SophieDePaula) [inline pending-changes.yaml:65]: @benbabayoff8

**Comment by SophieDePaula** [comment]
@kyryloonufriiev please see the latest changes

File: js/packages/eyes-cypress/src/plugin/pluginExport.js [hunk 2]
```diff
@@ -5,30 +5,35 @@ const makeGlobalRunHooks = require('./hooks');
...
       const {server, port, closeManager, closeBatches, closeUniversalServer} = await startServer();
       eyesServer = server;
 
       const globalHooks = makeGlobalRunHooks({closeManager, closeBatches, closeUniversalServer});
 
       const [origOn, config] = args;
+
+      if (!pluginModuleExports) {
+        pluginModuleExports =
+          config.testingType === 'e2e' ? pluginExportsE2E : pluginExportsComponent;
```
Comment (kyryloonufriiev) [inline js/packages/eyes-cypress/src/plugin/pluginExport.js:34]: Where this value will come from? Is it something cypress adds on itself?

File: js/packages/eyes-cypress/src/plugin/pluginExport.js [hunk 3]
```diff
@@ -5,30 +5,35 @@ const makeGlobalRunHooks = require('./hooks');
...
-        pluginModuleExports = pluginModule.exports.default;
-      }
-    } else if (pluginModule.exports.e2e) {
-      pluginModuleExports = pluginModule.exports.e2e.setupNodeEvents;
-    } else if (pluginModule.exports.component) {
-      pluginModuleExports = pluginModule.exports.component.setupNodeEvents;
-    } else {
-      pluginModuleExports = pluginModule.exports;
+    let eyesServer, pluginModuleExports, pluginExportsE2E, pluginExportsComponent;
+    let pluginExports =
```
Comment (kyryloonufriiev) [inline js/packages/eyes-cypress/src/plugin/pluginExport.js:9]: Why is that `let`? You are not going to change it, or am I missing something?

File: js/packages/eyes-cypress/src/plugin/pluginExport.js (hunk 2 above)
Comment (kyryloonufriiev) [reply js/packages/eyes-cypress/src/plugin/pluginExport.js:34]: If there is indeed a value that tells you what of those two you really care about (`e2e` or `component`), then you could simplify the implementation above, by not mapping both to specific variables.

Comment (SophieDePaula) [reply js/packages/eyes-cypress/src/plugin/pluginExport.js:34]: @kyryloonufriiev I do have the info about the test type, but it is only available when `setupNodeEvents` is called after I overwrite it.
When we start cypress our plugin code is called, then we overwrite the  `setupNodeEvents` to execute both our hooks and the user's hooks.
Then, the user chooses if they want to execute `e2e` or `component` testing, and at that point, the corresponding  `setupNodeEvents` method is executed. Only at that point, do I have the information the info about the test type.

File: js/packages/eyes-cypress/src/plugin/pluginExport.js (hunk 3 above)
Comment (SophieDePaula) [reply js/packages/eyes-cypress/src/plugin/pluginExport.js]: done

File: js/packages/eyes-cypress/src/plugin/pluginExport.js (hunk 2 above)
Comment (kyryloonufriiev) [reply js/packages/eyes-cypress/src/plugin/pluginExport.js:34]: Ok get it

**Review by kyryloonufriiev — APPROVED** [review]
You can merge it

**Pull request merged**